account_invoice_refund_reason/wizard/account_invoice_refund.py

```diff
@@ -0,0 +1,132 @@
+                        invoice.update({
+                            'type': inv.type,
+                            'date_invoice': form.date_invoice,
+                            'state': 'draft',
+                            'number': False,
+                            'invoice_line_ids': invoice_lines,
+                            'tax_line_ids': tax_lines,
+                            'date': date,
+                            'origin': inv.origin,
+                            'fiscal_position_id': inv.fiscal_position_id.id,
+                        })
+                        for field in inv_obj._get_refund_common_fields():
+                            if inv_obj._fields[field].type == 'many2one':
+                                invoice[field] = invoice[field] and\
+                                    invoice[field][0]
+                            else:
+                                invoice[field] = invoice[field] or False
+                        inv_refund = inv_obj.create(invoice)
+                        body = _('Correction of <a href=# \
+                            data-oe-model=account.invoice data-oe-id=%d>%s</a>\
+                            <br>Reason: %s') % (
+                            inv.id, inv.number, description)
+                        inv_refund.message_post(body=body)
+                        if inv_refund.payment_term_id.id:
+                            inv_refund._onchange_payment_term_date_invoice()
+                        created_inv.append(inv_refund.id)
+                xml_id = inv.type == 'out_invoice' and\
+                    'action_invoice_out_refund' or \
+                    inv.type == 'out_refund' and 'action_invoice_tree1' or \
+                    inv.type == 'in_invoice' and 'action_invoice_in_refund' or\
+                    inv.type == 'in_refund' and 'action_invoice_tree2'
+        if xml_id:
+            result = self.env.ref('account.%s' % (xml_id)).read()[0]
+            if mode == 'modify':
+                # When refund method is `modify` then it will directly open\
+                # the new draft bill/invoice in form view
+                if inv_refund.type == 'in_invoice':
+                    view_ref = self.env.ref('account.invoice_supplier_form')
+                else:
+                    view_ref = self.env.ref('account.invoice_form')
+                form_view = [(view_ref.id, 'form')]
+                if 'views' in result:
+                    result['views'] = form_view + \
+                        [(state, view)
+                         for state, view in result['views'] if view != 'form']
+                else:
+                    result['views'] = form_view
+                result['res_id'] = inv_refund.id
+            else:
+                invoice_domain = safe_eval(result['domain'])
+                invoice_domain.append(('id', 'in', created_inv))
+                result['domain'] = invoice_domain
+            return result
+        return True
```

Comment:
@Chanakya-SerpentCS Instead of copy/pasting a complete method without calling `super()`, I suggest you extend invoice_refund:
* call `super()`
* find the created refund
* set the reason_id field